acsylla/_cython/result/result.pyx

```diff
@@ -61,14 +61,26 @@ cdef class Result:
         count = cass_result_column_count(self.cass_result)
         return count
 
+    def columns_names(self):
+        """ Returns the columns names"""
+        cdef size_t length = 0
+        cdef char* name = NULL
+        cdef CassError error
+        columns = []
+        for i in range(self.column_count()):
+            error = cass_result_column_name(self.cass_result, i, <const char**> &name, <size_t*> &length)
+            raise_if_error(error)
+            columns.append(name.decode())
```

Comment:
do we need to deallocate the `name` ? I should compile to see if the `name.decode()` takes the ownership, but IIRC i would say that no

Reply:
@pfreixes  Thanks for review 
I'm not sure, but how I can fix it?

Reply:
we need to check if `cass_result_column_name` allocates memory for this, and if this is the case we need to free [1] the pointer after doing the encode. Could be that `cass_result_column_name` just returns a pointer to an internal already existent data slice, if this is the case the release should happen transparently when the result object is deleted

[1] https://cython.readthedocs.io/en/latest/src/tutorial/memory_allocation.html

Reply:
Ah, i see, ok! thanks

Reply:
@pfreixes maybe I wrong but looks like in C++ we do not need deallocate `name`

Reply:
any reference that I can take a look?

Reply:
https://cython.readthedocs.io/en/latest/src/tutorial/strings.html#c-strings

When i try to free  `name` i got:
Python (58506,0x117f48e00) malloc: *** error for object 0x7f88ebc210ac: A deallocated pointer was not allocated

Reply:
@pfreixes BTW yesterday I maked test in real enviroment on table with 86 colums with select 1B rows and have no see memory leak at this function. But 
Also I tested inserts ~100M rows (with bind_dict on prepared statement) in same table and don't have memory leek. Looks like all is good